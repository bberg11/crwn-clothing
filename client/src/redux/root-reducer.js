import { combineReducers } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import UserReducer from './user/user.reducer';
import CartReducer from './cart/cart.reducer';
import DirectoryReducer from './directory/directory.reducer';
import ShopReducer from './shop/shop.reducer';
import OrdersReducer from './orders/orders.reducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['cart', 'orders'],
};

const rootReducer = combineReducers({
  user: UserReducer,
  cart: CartReducer,
  directory: DirectoryReducer,
  shop: ShopReducer,
  orders: OrdersReducer,
});

export default persistReducer(persistConfig, rootReducer);
