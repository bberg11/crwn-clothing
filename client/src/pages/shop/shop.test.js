import React from 'react';
import { mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import Shop from './shop.component';

export const createMockStore = ({ state, reducers }) => {
  const store = createStore(combineReducers(reducers), state);
  return {
    ...store,
    persistor: {
      persist: () => null,
    },
  };
};

describe('Shop Page', () => {
  let wrapper;
  let mockFetchCollectionsStart;

  beforeEach(() => {
    mockFetchCollectionsStart = jest.fn();

    const mockMatch = {
      path: '',
    };

    const mockProps = {
      match: mockMatch,
      fetchCollectionsStart: mockFetchCollectionsStart,
    };

    // wrapper = mount(
    //   <BrowserRouter>
    //     <Shop {...mockProps} />
    //   </BrowserRouter>
    // );
  });

  it('should render Shop Page component', () => {
    // expect(wrapper).toMatchSnapshot();
  });

  // it('should fire fetchCollectionsStart', () => {
  //   expect(mockFetchCollectionsStart).toHaveBeenCalled();
  // });
});
