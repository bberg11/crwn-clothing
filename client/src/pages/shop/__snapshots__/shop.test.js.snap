// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Page should render Shop Page component 1`] = `ReactWrapper {}`;
